### ISSUE TITLE
Convert .ply to .splat

### ISSUE THREAD

**Issue by imbahhc** [mkkellogg/GaussianSplats3D#28]
Dear author,
I'm a new user for java. Could you please tell me where to put the code you provide to convert .ply file to .splat file? Thank you.
Best regards,
hhc

**Comment by mkkellogg** [comment]
That's a very good question, right now I just run the code from inside a web page. I should really make a node script to do the conversion -- I'll work on that :) For now, just create a web page that looks like this and open it:
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <script type="importmap">
      {
          "imports": {
              "three": "./lib/three.module.js",
              "gaussian-splats-3d": "./lib/gaussian-splat-3d.module.js"
          }
      }
    </script>
  </head>
  <body>
    <script type="module">
      import * as GaussianSplats3D from 'gaussian-splats-3d';
      const compressionLevel = 1;
      const splatAlphaRemovalThreshold = 10;
      const plyLoader = new GaussianSplats3D.PlyLoader();
      plyLoader.loadFromFile('<path to .ply file>', compressionLevel, splatAlphaRemovalThreshold)
      .then((splatBuffer) => {
          new GaussianSplats3D.SplatLoader(splatBuffer).saveToFile('converted_file.splat');
      });
    </script>
  </body>
</html>
```

**Comment by mkkellogg** [comment]
Not sure if you're still interested, but the latest version has a UI for converting `.ply` files to `.splat` at [http://localhost:8080/index.html](http://localhost:8080/index.html)

**Comment by softyoda** [comment]
Hi quick question relating to .splat format.

Are you still using same .spat format than other integration (aframe/antimatter)?

In your readme you say "currently only compatible with this viewer", does that mean your .splat format is more optimized, and you integrate ways to reduce the height of a scene using Aras optimization such as chunking / texture / clustering / spherical-harmonics lower order precision to have less bit per splat ?

**Comment by mkkellogg** [comment]
Yes I have implemented some of my own optimizations to reduce the size of the `.splat` file. Currently I have implemented chunking to a small extent as well as the option to store position/scale/rotation values as 16-bit instead of 32-bit. Optimizing the `.splat` format is still an ongoing project; I know there are still several things I can do. I have read through Aras's blog on reducing file size and I will hopefully get around to implementing more of those ideas soon.
